apps/services/auth/delegation-api/src/app/domains/test/domains.controller.spec.ts

```diff
@@ -0,0 +1,385 @@
+          user: testCase.user,
+          customScopeRules: testCase.customScopeRules,
+        })
+        server = request(app.getHttpServer())
+
+        const factory = new FixtureFactory(app)
+        await Promise.all(
+          testCase.domains.map((domain) => factory.createDomain(domain)),
+        )
+        await Promise.all(
+          (testCase.accessTo ?? []).map((scope) =>
+            factory.createApiScopeUserAccess({
+              nationalId: testCase.user.nationalId,
+              scope,
+            }),
+          ),
+        )
+        await Promise.all(
+          (testCase.delegations ?? []).map((delegation) =>
+            factory.createCustomDelegation({
+              fromNationalId: testCase.user.nationalId,
+              toNationalId: testCase.user.actor?.nationalId,
+              ...delegation,
+            }),
+          ),
+        )
+      })
+
+      it('GET /domains returns expected domains', async () => {
+        // Arrange
+        const expected = testCase.expected.map(({ name }) => ({
+          name,
+        }))
+
+        // Act
+        const res = await server.get('/v1/domains')
+
+        // Assert
+        expect(res.status).toEqual(200)
+        expect(res.body).toHaveLength(testCase.expected.length)
+        expect(res.body).toMatchObject(expected)
+      })
+
+      if (validDomains.length) {
+        it.each(validDomains)(
+          'GET /domains/%s returns expected domains',
+          async (domainName) => {
+            // Arrange
+            const domain = testCase.expected.find(
+              (domain) => domain.name === domainName,
+            )
+            assert(domain)
+            const expected = { name: domain.name }
+
+            // Act
+            const res = await server.get(`/v1/domains/${domainName}`)
+
+            // Assert
+            expect(res.status).toEqual(200)
+            expect(res.body).toMatchObject(expected)
+          },
+        )
+
+        it.each(validDomains)(
+          'GET /domains/%s/scopes returns expected scopes',
+          async (domainName) => {
+            // Arrange
+            const domain = testCase.expected.find(
+              (domain) => domain.name === domainName,
+            )
+            assert(domain)
+            const expected = domain.scopes
+
+            // Act
+            const res = await server.get(`/v1/domains/${domainName}/scopes`)
+
+            // Assert
+            expect(res.status).toEqual(200)
+            expect(res.body).toHaveLength(expected.length)
+            expect(res.body).toMatchObject(expected)
+          },
+        )
+
+        it.each(validDomains)(
+          'GET /domains/%s/scope-tree returns expected scope tree',
+          async (domainName) => {
+            // Arrange
+            const domain = testCase.expected.find(
+              (domain) => domain.name === domainName,
+            )
+            assert(domain)
+            const expected = domain.scopes
```

Comment:
Yeah, I decided to focus the test case "DSL" on access checks, and do custom tests for other logic. Otherwise the DSL and tests might become unnecessarily complicated.